nix/sources.nix

```diff
@@ -1,8 +1,56 @@
 # This file has been generated by Niv.
 
-# A record, from name to path, of the third-party packages
-with rec
-{
+let
+
+  #
+  # The fetchers. fetch_<type> fetches specs of type <type>.
+  #
+
+  fetch_file = spec:
+    if spec.builtin or true then
+      builtins_fetchurl { inherit (spec) url sha256; }
+    else
+      pkgs.fetchurl { inherit (spec) url sha256; };
+
+  fetch_tarball = spec:
+    if spec.builtin or true then
```

Comment:
@nmattia would you mind explaining this one by any chance? Is this some sort of nix syntax trick? I am across this whilst wondering why my tarball (zip) wasn't extracted after download

Reply:
The `spec`, i.e. the package in the `sources.json` has an optional `builtin` field. When set to `true`, then it doesn't use the nixpkgs fetchers, but instead uses the `builtin` fetchers. By default, as seen here, the default behavior is `true`, i.e. _do_ use the built-in fetchers.

You'll want the built-in fetcher if you do _not_ want `nixpkgs` to be evaluated. You _do_ want the built-in fetchers if you're using nix with restricted eval mode, like e.g. on Hydra. Hope this helps!

Reply:
Thank you! Sorry, what I meant is: doesn't this always take the first branch? Because `or true` always evaluates to true? Am i meant to manually change the initially generated nix file?

Reply:
oh, the `or` syntax means "either use the attribute (`.builtin` here) _or_ use this other value if the attribute isn't set".